### PR TITLE
ENH: Print object arrays containing lists unambiguously

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously

    array([[1], [1, 1], [1, 1], [1]], dtype=object)

Now

    array([list([1]), list([1, 1]), list([1, 1]), list([1])], dtype=object)

### CODE REVIEW

**Comment by ahaldane** [comment]
+1 from me. Code LGTM.

**Comment by eric-wieser** [comment]
Updated with a release note, that also includes an entry about #8963, since they're related

File: numpy/core/arrayprint.py [hunk 1]
```diff
@@ -256,6 +264,7 @@ def _get_formatdict(data, precision, suppress_small, formatter):
                   'longcomplexfloat': lambda: LongComplexFormat(precision),
                   'datetime': lambda: DatetimeFormat(data),
                   'timedelta': lambda: TimedeltaFormat(data),
+                  'object': lambda: _object_format,
                   'numpystr': lambda: repr_format,
                   'str': lambda: str}
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: Interesting fact - this is never nor was ever used...

**Comment by eric-wieser** [comment]
Adding to the milestone since this contains a release note fix for an already-included feature

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**